### ISSUE TITLE
Date converter does not work when running on Robolectric

### ISSUE THREAD

**Issue by h6ah4i** [bluelinelabs/LoganSquare#51]
Actually, this is not an issue of LoganSquare. I noticed that the DefaultDateConverter does not work when running on Robolectric.

### Running on Android:
SimpleDateFormat accepts these format for the "Z" place holder: `Z/ZZ/ZZZ:-0800 ZZZZ:GMT-08:00 ZZZZZ:-08:00`
ref.) http://developer.android.com/reference/java/text/SimpleDateFormat.html

### Running on Robolectric (JRE):
SimpleDateFormat only accepts "-0800" format for the "Z" place holder. Should use "X" place holder for ISO 8601 format.
ref.) http://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html#rfc822timezone

---

### Proposed fix:

Use `+0000` in the `DefaultDateFormatter.getFixedInputString()` method. This works on both Android and JRE.

```java
public class DefaultDateFormatter extends SimpleDateFormat {
    /** Replace ending Z's with +0000 so Java's SimpleDateFormat can handle it correctly */
    private String getFixedInputString(String input) {
        return input.replaceAll("Z$", "+0000");
    }
}
```

**Comment by EricKuck** [comment]
Sorry about the wait! I've made this change and will be merging into master over the weekend. I'll post another comment when it's in.

**Comment by EricKuck** [comment]
This has now been pushed to master in 7d6f2e6f6c0bd10e31b3fa9fce112e8cb0fb252c and is available on JCenter as version 1.1.0.
